src/script/lua_api/l_localplayer.cpp

```diff
@@ -223,16 +223,17 @@ int LuaLocalPlayer::l_get_control(lua_State *L)
 	};
 
 	lua_createtable(L, 0, 12);
-	set("up", c.up);
-	set("down", c.down);
-	set("left", c.left);
-	set("right", c.right);
-	set("jump", c.jump);
-	set("aux1", c.aux1);
+	set("jump",  c.jump);
+	set("aux1",  c.aux1);
 	set("sneak", c.sneak);
-	set("zoom", c.zoom);
-	set("dig", c.dig);
+	set("zoom",  c.zoom);
+	set("dig",   c.dig);
 	set("place", c.place);
+	// Legacy field to ensure compatibility
+	set("up",    player->keyPressed & (1 << 0)); // Up, down, left, and right were removed in favor of
+	set("down",  player->keyPressed & (1 << 1)); // analog  direction indicators and are therefore not
+	set("left",  player->keyPressed & (1 << 2)); // available as booleans anymore. The corresponding values
+	set("right", player->keyPressed & (1 << 3)); // can still be read from the keyPressed bits though.
```

Comment:
It would make sense to expose `movement_speed` and `movement_direction` here.
Also since this is CSM we have no compatibility guarantee but keeping up,down,left,right is still a good idea.

Reply:
exposing them now, I don't know if the handling on the `l_object.cpp` side is correct though